### PR TITLE
Added script to store user specific activity cards to gold table

### PR DESCRIPTION
**Pull request by rohandudam**
Added a script to fetch all cards from silver unique cards table, filter data for specific user over specified time frame and save it to user Delta Lake Gold table.

Command to run the script:
`python user_activity_cards_gold.py --start_date "2023-01-01" --end_date "2023-02-20" --user_name "<user_name>"`

@sravantit25 FYI: Need better way to pass the path, Presently script works only for my user_name.

### CODE REVIEW

**Comment by sravantit25** [comment]
> python user_activity_cards_gold.py --start_date "2023-01-01" --end_date "2023-02-20" --user_name "<user_name>"

@r

> Added a script to fetch all cards from silver unique cards table, filter data for specific user over specified time frame and save it to user Delta Lake Gold table.
> 
> Command to run the script: `python user_activity_cards_gold.py --start_date "2023-01-01" --end_date "2023-02-20" --user_name "<user_name>"`
> 
> @sravantit25 FYI: Need better way to pass the path, Presently script works only for my user_name.

@rohandudam The script should be specific to your name. I had mentioned that in the instructions that the script should have your name hardcoded. Each person will have their own gold table and their own script to fetch the gold table. So, please update the script so that it takes your name by default. We need not have to pass it as argument. The idea is to have everyone familiarized with Delta tables.

File: jobs/user_activity_card_gold.py [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+import sys
+from datetime import datetime
+import argparse
+from loguru import logger
+from pyspark.sql.functions import col
+sys.path.append(os.path.dirname(os.path.dirname(os.path.abspath(__file__))))
+from helpers import spark_helper as sh
+from config import delta_table_conf as dc
+
+def get_user_cards_and_save_to_gold_table(user_name, start_date, end_date, path):
```
Comment (sravantit25) [inline jobs/user_activity_card_gold.py]: please default the user_name to your name and remove the path as the argument.

File: jobs/user_activity_card_gold.py [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+        # Fetch the cards data from the unique cards silver table
+        unique_silver_cards_data = spark.read.format("delta").load(dc.cards_unique_silver_table_path)
+        logger.info(f'Fetched the cards data from unique silver cards table')
+        
+        # Filter user card data for specified time period
+        filtered_users_unique_cards_data = unique_silver_cards_data.filter(col("card_members").contains(user_name) 
+            & (col("dateLastActivity") >= start_date) & (col("dateLastActivity") <= end_date))
+
+        # Select columns
+        filtered_users_unique_cards_data = filtered_users_unique_cards_data.select("id", "name", "LastUpdated", "board_name")
```
Comment (sravantit25) [inline jobs/user_activity_card_gold.py]: Please fetch card_members too here.

File: jobs/user_activity_card_gold.py [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+    except Exception as error:
+        logger.exception(
+            f'Exception while fetching data for the user: {user_name}')
+        raise error
+
+
+# ---START OF SCRIPT----
+if __name__ == "__main__":
+    PARSER = argparse.ArgumentParser(description='Create Individual gold card table')
+    PARSER.add_argument('--user_name', metavar='Please pass the user_name', required=True,
```
Comment (sravantit25) [inline jobs/user_activity_card_gold.py]: This argument need not have to be passed. Default it to your name.

File: jobs/user_activity_card_gold.py [hunk 4]
```diff
@@ -0,0 +1,69 @@
+"""
```
Comment (sravantit25) [inline jobs/user_activity_card_gold.py]: Please change the script name to rohan_cards_gold.py. Each person will have their own table and needs to be uniquely identifiable.

**Review by sravantit25 — CHANGES_REQUESTED** [review]

File: jobs/user_activity_card_gold.py (hunk 4 above)
Comment (rohandudam) [reply jobs/user_activity_card_gold.py]: @sravantit25 I am done with all suggested changes. I re-based my branch with master so you will notice multiple file changes, please look at latest commit for your simplicity.

Comment (sravantit25) [reply jobs/user_activity_card_gold.py]: @rohandudam Thank you Rohan. Looks good. Merged the PR.

**Pull request merged**